test-packages/test-utils/src/utils/semantic-model-provider.ts

```diff
@@ -80,15 +188,21 @@ export function loadLibraries(version: TestModelVersion): Record<string, Json> {
   return libToFileContent;
 }
 
-export function generateModel(
+export async function generateModel(
   version: TestModelVersion,
+  downloadLibs = true,
   disableCache = false
-): UI5SemanticModel {
+): Promise<UI5SemanticModel> {
   if (has(MODEL_CACHE, version) && !disableCache) {
     return MODEL_CACHE[version];
   }
 
+  if (downloadLibs) {
+    await downloadLibraries(version);
```

Comment:
Are we assuming here that the test infrastructure never executes test N+1 before test N has finished ?

Could subsequent calls to `generateModel` be executed before the first one finished?
and the libraries be downloaded multiple times?

I am not sure this can happen in Mocha, but perhaps we should take care of this regardless? (e.g return a **singleton** promise to the result (per ui5 version) instead (?) of a **new** promise each time this function is called

Reply:
Changed in `downloadLibraries`